### ISSUE TITLE
Are you still working on this?

### ISSUE THREAD

**Issue by miguelcarcamov** [pierrepaleo/PDWT#1]
Hi there,

I am interested on working with this API since it is very useful for my thesis work. Do you know if the functions are compatible with CUDA 10?

Best regards

**Comment by pierrepaleo** [comment]
Hi @miguelcarcamov 

This project has not been updated for a while, but as far as I know things are working.
I just tested the code on a machine with CUDA 10 and all [pypwt](https://github.com/pierrepaleo/pypwt) tests passed.

You will certainly need to modify the `-arch` compilation flag in `PDWT/Makefile` or `pypwt/setup.py`. For example I used `-arch=sm_70` for a Tesla V100-SXM2.

The [pypwt](https://github.com/pierrepaleo/pypwt) python wrapper runs numerous tests against the [pywavelets](https://pypi.org/project/PyWavelets) module. However the API of pywavelets changed (notably for SWT), so tests fail with recent versions of pywavelets. The version `0.5.0` has to be used.

I tested the following with Python 3.6 in a virtual environment:

```bash
git clone https://github.com/pierrepaleo/pypwt
cd pypwt
# Modify the -arch compilation flag in setup.py depending on your device
python setup.py build
python setup.py install
pip install pywavelets==0.5.0 # see above comments 
python test/test_all.py
```

Tests have to be updated to compare against recent version of pywavelets.

**Comment by miguelcarcamov** [comment]
what if I want to use it just in C++/CUDA? It should work despite my version of pywavelets right?

**Comment by lydiazajiczek** [comment]
Hi Pierre,

Sorry to hijack your response to the other issue but I have been meaning to email you for ages!

First of all thanks for this great library. I unfortunately ran into a very strange behaviour that I was never able to figure out and wondered if you had any insight.

I use your library to run the two dimensional wavelet transform on a stack of images, then I find the maximum wavelet coefficient for each pixel along z and do an inverse transform on the composite/maximum wavelet coefficients. The problem is, running the transform/inverse transform is non deterministic and the result looks different every time. This made me think it was a race condition - CUDA has a race condition checker  but it didn't find anything.

Have you ever seen this behaviour with your library?

Thanks a lot!

Lydia Neary-Zajiczek

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by pierrepaleo** [comment]
@miguelcarcamov Yes pywavelets is just for testing the Python wrapper, so the C++/CUDA API will work without it. What I meant is that if the Python tests pass, then the underlying C++/CUDA code should be valid. I also ran the bare demo of PDWT without issues.

@lydiazajiczek Thanks for the report ; would you be able to provide a minimal code reproducing this behavior ? As far as I remember, the only cases when I got these behaviors is when an illegal memory access is done on the image/coefficients (ex. buggy thresholding function).
